h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -0,0 +1,143 @@
+'use strict';
+
+var Controller = require('../base/controller');
+var setElementState = require('../util/dom').setElementState;
+
+/**
+ * Controller for the search bar.
+ */
+class SearchBarController extends Controller {
+  constructor(element) {
+    super(element);
+
+    this._input = this.refs.searchBarInput;
+    this._dropdown = this.refs.searchBarDropdown;
+    this._dropdownItems = element.querySelectorAll('[data-ref="searchBarDropdownItem"]');
+
+    var getActiveElementIndex = () => {
+      return Array.prototype.slice.call(this._dropdownItems).
+        indexOf(element.querySelector('.js-search-bar-dropdown-menu-item--active'));
+    };
+
+    var clearActiveDropdownItem = () => {
+      var activeItem = element.querySelector('.js-search-bar-dropdown-menu-item--active');
+      if (activeItem) {
+        activeItem.classList.remove('js-search-bar-dropdown-menu-item--active');
+      }
+    };
+
+    var updateActiveDropdownItem = index => {
+      clearActiveDropdownItem();
+      this._dropdownItems[index].classList.add('js-search-bar-dropdown-menu-item--active');
+    };
+
+    var onSelectFacet = facet => {
```

Comment:
I don't think this should be an `on*()` function because it's not a function that is called automatically on an event (e.g. `onEnterKeyPressed()` or something), rather it's called explicitly by our code in a couple of places, so I'd name the function after what it does, maybe just `selectFacet()`.